### PR TITLE
dotCMS#23623 Experiment Split Traffic 

### LINKED ISSUE #23623: [BE] Change Split Traffic
image

image

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Allow to change the split traffic over the different variants 

https://user-images.githubusercontent.com/31667212/218890214-e2884aad-aaf0-4f28-b9b0-4f222a4776a2.mov

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.html [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                novalidate
+            >
+                <div class="field">
+                    <label>
+                        {{ 'experiments.configure.traffic.split.name' | dm }}
+                    </label>
+                    <p-radioButton
+                        [label]="'experiments.configure.traffic.split.evenly' | dm"
+                        (click)="splitEvenly()"
+                        value="SPLIT_EVENLY"
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.html:22]: Use the constant Enum

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.html [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                        formControlName="type"
+                    >
+                    </p-radioButton>
+                    <p-radioButton
+                        [label]="'experiments.configure.traffic.split.custom' | dm"
+                        value="CUSTOM_PERCENTAGES"
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.html:28]: Same.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.html [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                        >
+                            <span
+                                class="experiment-traffic-split__variants-name"
+                                data-testId="variant-name"
+                            >
+                                {{ variant.value.name }}
+                            </span>
+                            <p-inputNumber
+                                class="experiment-traffic-split__variants-weight"
+                                *ngIf="form.value.type === 'CUSTOM_PERCENTAGES'; else splitEven"
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.html:53]: Use the constant

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic/dot-experiments-configuration-traffic.component.ts [hunk 4]
```diff
@@ -50,18 +56,24 @@ export class DotExperimentsConfigurationTrafficComponent {
...
+        | DotExperimentsConfigurationTrafficAllocationAddComponent
+        | DotExperimentsConfigurationTrafficSplitAddComponent
+    >;
+    private trafficConfig: TrafficConfig;
 
     constructor(
         private readonly dotExperimentsConfigurationStore: DotExperimentsConfigurationStore
     ) {}
 
     changeTrafficAllocation() {
```
Comment (alfredo-dotcms) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic/dot-experiments-configuration-traffic.component.ts:69]: doc

**Review by fmontes — DISMISSED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.html (hunk 1 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.html]: done

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        expect(submitButton.disabled).toBeFalse();
+        expect(submitButton).toContainText('Done');
+        expect(spectator.component.form.valid).toBeTrue();
+
+        spectator.click(submitButton);
+        expect(store.setSelectedTrafficProportion).toHaveBeenCalledWith({
+            trafficProportion: ExperimentMocks[1].trafficProportion,
+            experimentId: EXPERIMENT_ID
+        });
+    });
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.spec.ts:97]: a message handled in the store.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+                    nonNullable: true,
+                    validators: [Validators.required]
+                }),
+                variants: this.fb.array<Variant>([])
+            });
+
+            data.trafficProportion.variants.forEach((variant) => {
+                this.variants.push(this.addVariantToForm(variant));
+            });
+        });
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.ts:130]: just once, to init de form.

**Review by alfredo-dotcms — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.spec.ts:97]: Ok, make sure you check that in the store.

**Pull request merged**